### ISSUE TITLE
Windows (appveoyr) issues

### ISSUE THREAD

**Issue by rc** [sfepy/sfepy#427]
Appveyor tests that fail:

In Python 3.6:
```
Environment: PYTHON=C:\Python36_64, PYTHON_VERSION=3.6, PYTHON_ARCH=64, CONDA_PY=36
```
```
[12] tests\test_homogenization_perfusion.py
---   test_solution: failed!
[34] tests\test_input_linear_elastic_mM.py
---   test_input: failed!
[42] tests\test_input_navier_stokes2d_iga.py
---   test_input: failed!
[79] tests\test_meshio.py
+++   test_read_meshes: ok
+++   test_compare_same_meshes: ok
+++   test_read_dimension: ok
+++   test_write_read_meshes: ok
---   test_hdf5_meshio: failed!
[88] tests\test_projections.py
+++   test_mass_matrix: ok
+++   test_project_tensors: ok
---   test_projection_iga_fem: failed!
+++   test_projection_tri_quad: ok
```
[42, 88] are due to missing igakit and could be disabled on Windows. What about the others?

[12] fails with:
```
<<< [12] tests\test_homogenization_perfusion.py
[12] sfepy: left over: ['__name__', '__doc__', '__package__', '__loader__', '__spec__', '__file__', '__cached__', '__builtins__', 'print_function', 'absolute_import', 'six', 'input_name', 'TestCommon', 'verbose', '_filename']
>>> test instance prepared (1 test(s))
../examples/homogenization/perfusion_micro.py
[12] sfepy: left over: ['__name__', '__doc__', '__package__', '__loader__', '__spec__', '__file__', '__cached__', '__builtins__', 'absolute_import', 'match_x_plane', 'match_y_plane', 'cb', 'nm', 'data_dir', 'six', 'range', 'get_mats', 'recovery_perf', 'geoms', 'pb_def', 'eps0', 'param_h', 'aux', 'ch', 'val', 'all_periodicYM', 'all_periodicY', 'reg_io', 'ebcs_eta', 'ebcs_gamma', 'ch2', 'val2', 'aux_bY', 'i_io', 'io', 'ion', 'matk1', 'matk2', 'ipm', 'ireg', 'coefs', 'requirements', 'ipm2', 'get_channel', 'set_corrpis', 'set_corr_S', 'set_corr_cc', 'i_io2', 'io2', 'io12', 'verbose', '_filename']
homogen: reading mesh [line2, tri3, quad4, tetra4, hexa8] (C:\projects\sfepy\meshes\3d\perfusion_micro3d.mesh)...
homogen: ...done in 0.08 s
homogen: creating regions...
homogen:     Y
homogen:     YM
homogen:     Pl
homogen:     Pr
homogen:     Pn
homogen:     Pf
homogen:     bYm
homogen:     bYp
homogen:     YA
homogen:     fixedYA
homogen:     YB
homogen:     fixedYB
homogen:     PlYM
homogen:     PrYM
homogen:     PnYM
homogen:     PfYM
homogen:     bYMA
homogen:     PlYA
homogen:     PrYA
homogen:     PnYA
homogen:     PfYA
homogen:     bYA_1
homogen:     bYA_2
homogen:     bYA_3
homogen:     bYMB
homogen:     PlYB
homogen:     PrYB
homogen:     PnYB
homogen:     PfYB
homogen:     bYB_1
homogen:     bYB_2
homogen:     bYB_3
homogen:     bYMp
homogen:     bYMm
homogen:     bYA
homogen:     bYB
homogen:     bYMpm
homogen:     bYMchs
homogen:     bYA_n1
homogen:     bYA_n2
homogen:     bYA_n3
homogen:     bYB_n1
homogen:     bYB_n2
homogen:     bYB_n3
homogen:     YMmchs
homogen: ...done in 0.16 s
homogen: using solvers:
                ts: no ts
               nls: newton
                ls: ls
>>> <class 'TypeError'>
python : Traceback (most recent call last):
At line:2 char:28
+ if ($LastExitCode -ne 0) { python run_tests.py --raise }
+                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (Traceback (most recent call last)::String) [], RemoteException
    + FullyQualifiedErrorId : NativeCommandError
 
  File "run_tests.py", line 234, in <module>
    sys.exit(main())
  File "run_tests.py", line 226, in main
    run_tests(stats, dirpath, filenames)
  File "run_tests.py", line 144, in run_tests
    stats[0] + 1)
  File "run_tests.py", line 108, in run_test
    ifile=ifile)
  File "C:\projects\sfepy\sfepy\base\testing.py", line 67, in run
    ret = test_method()
  File "C:\projects\sfepy\tests\test_homogenization_perfusion.py", line 51, in test_solution
    coefs = app()
  File "C:\projects\sfepy\sfepy\applications\application.py", line 29, in call_basic
    return self.call(**kwargs)
  File "C:\projects\sfepy\sfepy\homogenization\homogen_app.py", line 172, in call
    aux = self.he(ret_all=ret_all, time_tag=time_tag)
  File "C:\projects\sfepy\sfepy\applications\application.py", line 29, in call_basic
    return self.call(**kwargs)
  File "C:\projects\sfepy\sfepy\homogenization\engine.py", line 744, in call
    time_tag)
  File "C:\projects\sfepy\sfepy\homogenization\engine.py", line 375, in __call__
    w.start()
  File "C:\Python36_64\envs\sfepy-test\lib\multiprocessing\process.py", line 105, in start
    self._popen = self._Popen(self)
  File "C:\Python36_64\envs\sfepy-test\lib\multiprocessing\context.py", line 223, in _Popen
    return _default_context.get_context().Process._Popen(process_obj)
  File "C:\Python36_64\envs\sfepy-test\lib\multiprocessing\context.py", line 322, in _Popen
    return Popen(process_obj)
  File "C:\Python36_64\envs\sfepy-test\lib\multiprocessing\popen_spawn_win32.py", line 65, in __init__
    reduction.dump(process_obj, to_child)
  File "C:\Python36_64\envs\sfepy-test\lib\multiprocessing\reduction.py", line 60, in dump
    ForkingPickler(file, protocol).dump(obj)
TypeError: can't pickle module objects
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "C:\Python36_64\envs\sfepy-test\lib\multiprocessing\spawn.py", line 105, in spawn_main
    exitcode = _main(fd)
  File "C:\Python36_64\envs\sfepy-test\lib\multiprocessing\spawn.py", line 115, in _main
    self = reduction.pickle.load(from_parent)
EOFError: Ran out of input
Command executed with exception: Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "C:\Python36_64\envs\sfepy-test\lib\multiprocessing\spawn.py", line 105, in spawn_main
    exitcode = _main(fd)
  File "C:\Python36_64\envs\sfepy-test\lib\multiprocessing\spawn.py", line 115, in _main
    self = reduction.pickle.load(from_parent)
EOFError: Ran out of input
```

In Python 2.7:
```
Environment: PYTHON=C:\Python27_64, PYTHON_VERSION=2.7, PYTHON_ARCH=64, CONDA_PY=27
```
The code does not compile, see https://ci.appveyor.com/project/rc/sfepy/build/job/ipwb7fp956ukw2f6:
```
C:\Users\appveyor\AppData\Local\Programs\Common\Microsoft\Visual C++ for Python\9.0\VC\Bin\amd64\cl.exe /c /nologo /Ox /MD /W3 /GS- /DNDEBUG -D__SDIR__=\"sfepy\\mechanics\\extmods\" -DSFEPY_PLATFORM=1 -Dinline=__inline -Isfepy\mechanics\extmods -Isfepy\discrete\common\extmods -IC:\Python27_64\envs\sfepy-test\lib\site-packages\numpy\core\include -IC:\Python27_64\envs\sfepy-test\include -IC:\Python27_64\envs\sfepy-test\PC /Tcsfepy\mechanics\extmods\contres.c /Fobuild\temp.win-amd64-2.7\Release\sfepy\mechanics\extmods\contres.obj -g -O2
cl : Command line warning D9002 : ignoring unknown option '-g'
contres.c
sfepy\mechanics\extmods\contres.c(255) : error C2275: 'float64' : illegal use of this type as an expression
        c:\projects\sfepy\sfepy\discrete\common\extmods\types.h(16) : see declaration of 'float64'
sfepy\mechanics\extmods\contres.c(255) : error C2146: syntax error : missing ';' before identifier 'Xp'
sfepy\mechanics\extmods\contres.c(255) : error C2065: 'Xp' : undeclared identifier
sfepy\mechanics\extmods\contres.c(255) : error C2109: subscript requires array or pointer type
sfepy\mechanics\extmods\contres.c(256) : error C2275: 'float64' : illegal use of this type as an expression
        c:\projects\sfepy\sfepy\discrete\common\extmods\types.h(16) : see declaration of 'float64'
sfepy\mechanics\extmods\contres.c(256) : error C2146: syntax error : missing ';' before identifier 'Xg'
sfepy\mechanics\extmods\contres.c(256) : error C2065: 'Xg' : undeclared identifier
sfepy\mechanics\extmods\contres.c(256) : error C2109: subscript requires array or pointer type
sfepy\mechanics\extmods\contres.c(273) : error C2143: syntax error : missing ';' before 'const'
sfepy\mechanics\extmods\contres.c(277) : error C2143: syntax error : missing ';' before 'const'
sfepy\mechanics\extmods\contres.c(282) : error C2143: syntax error : missing ';' before 'type'
sfepy\mechanics\extmods\contres.c(283) : error C2065: 'els' : undeclared identifier
sfepy\mechanics\extmods\contres.c(284) : error C2065: 'IENrows' : undeclared identifier
sfepy\mechanics\extmods\contres.c(302) : error C2143: syntax error : missing ';' before 'const'
sfepy\mechanics\extmods\contres.c(305) : error C2143: syntax error : missing ';' before 'const'
sfepy\mechanics\extmods\contres.c(310) : error C2143: syntax error : missing ';' before 'type'
sfepy\mechanics\extmods\contres.c(311) : error C2065: 'elm' : undeclared identifier
sfepy\mechanics\extmods\contres.c(312) : error C2065: 'IENrowm' : undeclared identifier
sfepy\mechanics\extmods\contres.c(322) : error C2275: 'float64' : illegal use of this type as an expression
        c:\projects\sfepy\sfepy\discrete\common\extmods\types.h(16) : see declaration of 'float64'
sfepy\mechanics\extmods\contres.c(322) : error C2146: syntax error : missing ';' before identifier 'r'
sfepy\mechanics\extmods\contres.c(322) : error C2065: 'r' : undeclared identifier
sfepy\mechanics\extmods\contres.c(322) : warning C4244: '=' : conversion from 'float64' to 'int', possible loss of data
sfepy\mechanics\extmods\contres.c(323) : error C2275: 'float64' : illegal use of this type as an expression
        c:\projects\sfepy\sfepy\discrete\common\extmods\types.h(16) : see declaration of 'float64'
sfepy\mechanics\extmods\contres.c(323) : error C2146: syntax error : missing ';' before identifier 's'
sfepy\mechanics\extmods\contres.c(323) : error C2065: 's' : undeclared identifier
sfepy\mechanics\extmods\contres.c(323) : warning C4244: '=' : conversion from 'double' to 'int', possible loss of data
sfepy\mechanics\extmods\contres.c(326) : error C2065: 's' : undeclared identifier
sfepy\mechanics\extmods\contres.c(326) : warning C4244: '=' : conversion from 'float64' to 'int', possible loss of data
sfepy\mechanics\extmods\contres.c(332) : error C2065: 'r' : undeclared identifier
sfepy\mechanics\extmods\contres.c(335) : error C2065: 'r' : undeclared identifier
sfepy\mechanics\extmods\contres.c(335) : error C2065: 's' : undeclared identifier
sfepy\mechanics\extmods\contres.c(338) : error C2065: 'r' : undeclared identifier
sfepy\mechanics\extmods\contres.c(338) : error C2065: 's' : undeclared identifier
sfepy\mechanics\extmods\contres.c(341) : error C2065: 'r' : undeclared identifier
sfepy\mechanics\extmods\contres.c(341) : error C2065: 's' : undeclared identifier
sfepy\mechanics\extmods\contres.c(346) : error C2065: 'Xp' : undeclared identifier
sfepy\mechanics\extmods\contres.c(346) : error C2109: subscript requires array or pointer type
sfepy\mechanics\extmods\contres.c(347) : error C2065: 'Xg' : undeclared identifier
sfepy\mechanics\extmods\contres.c(347) : error C2109: subscript requires array or pointer type
sfepy\mechanics\extmods\contres.c(349) : error C2065: 'Xp' : undeclared identifier
sfepy\mechanics\extmods\contres.c(349) : error C2109: subscript requires array or pointer type
sfepy\mechanics\extmods\contres.c(350) : error C2065: 'Xg' : undeclared identifier
sfepy\mechanics\extmods\contres.c(350) : error C2109: subscript requires array or pointer type
sfepy\mechanics\extmods\contres.c(359) : error C2065: 'Xp' : undeclared identifier
sfepy\mechanics\extmods\contres.c(359) : error C2109: subscript requires array or pointer type
sfepy\mechanics\extmods\contres.c(359) : error C2065: 'Xg' : undeclared identifier
sfepy\mechanics\extmods\contres.c(359) : error C2109: subscript requires array or pointer type
sfepy\mechanics\extmods\contres.c(359) : error C2065: 'Xp' : undeclared identifier
sfepy\mechanics\extmods\contres.c(359) : error C2109: subscript requires array or pointer type
sfepy\mechanics\extmods\contres.c(359) : error C2065: 'Xg' : undeclared identifier
sfepy\mechanics\extmods\contres.c(359) : error C2109: subscript requires array or pointer type
sfepy\mechanics\extmods\contres.c(359) : error C2065: 'Xp' : undeclared identifier
sfepy\mechanics\extmods\contres.c(359) : error C2109: subscript requires array or pointer type
sfepy\mechanics\extmods\contres.c(359) : error C2065: 'Xg' : undeclared identifier
sfepy\mechanics\extmods\contres.c(359) : error C2109: subscript requires array or pointer type
sfepy\mechanics\extmods\contres.c(359) : error C2065: 'Xp' : undeclared identifier
sfepy\mechanics\extmods\contres.c(359) : error C2109: subscript requires array or pointer type
sfepy\mechanics\extmods\contres.c(359) : error C2065: 'Xg' : undeclared identifier
sfepy\mechanics\extmods\contres.c(359) : error C2109: subscript requires array or pointer type
sfepy\mechanics\extmods\contres.c(359) : error C2065: 'Xp' : undeclared identifier
sfepy\mechanics\extmods\contres.c(359) : error C2109: subscript requires array or pointer type
sfepy\mechanics\extmods\contres.c(359) : error C2065: 'Xg' : undeclared identifier
sfepy\mechanics\extmods\contres.c(359) : error C2109: subscript requires array or pointer type
sfepy\mechanics\extmods\contres.c(359) : error C2065: 'Xp' : undeclared identifier
sfepy\mechanics\extmods\contres.c(359) : error C2109: subscript requires array or pointer type
sfepy\mechanics\extmods\contres.c(359) : error C2065: 'Xg' : undeclared identifier
sfepy\mechanics\extmods\contres.c(359) : error C2109: subscript requires array or pointer type
sfepy\mechanics\extmods\contres.c(359) : error C2168: 'sqrt' : too few actual parameters for intrinsic function
sfepy\mechanics\extmods\contres.c(373) : error C2143: syntax error : missing ';' before 'const'
sfepy\mechanics\extmods\contres.c(374) : error C2065: 'dh' : undeclared identifier
sfepy\mechanics\extmods\contres.c(375) : error C2065: 'dh' : undeclared identifier
sfepy\mechanics\extmods\contres.c(381) : error C2275: 'float64' : illegal use of this type as an expression
        c:\projects\sfepy\sfepy\discrete\common\extmods\types.h(16) : see declaration of 'float64'
sfepy\mechanics\extmods\contres.c(381) : error C2146: syntax error : missing ';' before identifier 'Normal'
sfepy\mechanics\extmods\contres.c(381) : error C2065: 'Normal' : undeclared identifier
sfepy\mechanics\extmods\contres.c(381) : error C2109: subscript requires array or pointer type
sfepy\mechanics\extmods\contres.c(381) : error C2059: syntax error : '{'
sfepy\mechanics\extmods\contres.c(382) : error C2275: 'float64' : illegal use of this type as an expression
        c:\projects\sfepy\sfepy\discrete\common\extmods\types.h(16) : see declaration of 'float64'
sfepy\mechanics\extmods\contres.c(382) : error C2146: syntax error : missing ';' before identifier 'normal'
sfepy\mechanics\extmods\contres.c(382) : error C2065: 'normal' : undeclared identifier
sfepy\mechanics\extmods\contres.c(382) : error C2109: subscript requires array or pointer type
sfepy\mechanics\extmods\contres.c(382) : error C2059: syntax error : '{'
sfepy\mechanics\extmods\contres.c(384) : error C2065: 'Normal' : undeclared identifier
sfepy\mechanics\extmods\contres.c(384) : error C2109: subscript requires array or pointer type
sfepy\mechanics\extmods\contres.c(385) : error C2065: 'Normal' : undeclared identifier
sfepy\mechanics\extmods\contres.c(385) : error C2109: subscript requires array or pointer type
sfepy\mechanics\extmods\contres.c(386) : error C2065: 'Normal' : undeclared identifier
sfepy\mechanics\extmods\contres.c(386) : error C2109: subscript requires array or pointer type
sfepy\mechanics\extmods\contres.c(388) : error C2065: 'normal' : undeclared identifier
sfepy\mechanics\extmods\contres.c(388) : error C2109: subscript requires array or pointer type
sfepy\mechanics\extmods\contres.c(389) : error C2065: 'normal' : undeclared identifier
sfepy\mechanics\extmods\contres.c(389) : error C2109: subscript requires array or pointer type
sfepy\mechanics\extmods\contres.c(390) : error C2065: 'normal' : undeclared identifier
sfepy\mechanics\extmods\contres.c(390) : error C2109: subscript requires array or pointer type
sfepy\mechanics\extmods\contres.c(399) : error C2065: 'Normal' : undeclared identifier
sfepy\mechanics\extmods\contres.c(399) : error C2109: subscript requires array or pointer type
sfepy\mechanics\extmods\contres.c(400) : error C2065: 'Normal' : undeclared identifier
sfepy\mechanics\extmods\contres.c(400) : error C2109: subscript requires array or pointer type
sfepy\mechanics\extmods\contres.c(401) : error C2065: 'Normal' : undeclared identifier
sfepy\mechanics\extmods\contres.c(401) : error C2109: subscript requires array or pointer type
sfepy\mechanics\extmods\contres.c(403) : error C2143: syntax error : missing ';' before 'const'
sfepy\mechanics\extmods\contres.c(404) : error C2143: syntax error : missing ';' before 'const'
sfepy\mechanics\extmods\contres.c(405) : error C2143: syntax error : missing ';' before 'const'
sfepy\mechanics\extmods\contres.c(406) : error C2143: syntax error : missing ';' before 'const'
sfepy\mechanics\extmods\contres.c(407) : error C2143: syntax error : missing ';' before 'const'
sfepy\mechanics\extmods\contres.c(407) : fatal error C1003: error count exceeds 100; stopping compilation
warning: sfepy\discrete\common\extmods\_fmfield.pyx:6:31: Buffer unpacking not optimized away.
warning: sfepy\discrete\common\extmods\_fmfield.pyx:6:31: Buffer unpacking not optimized away.
warning: sfepy\discrete\common\extmods\_fmfield.pyx:19:31: Buffer unpacking not optimized away.
warning: sfepy\discrete\common\extmods\_fmfield.pyx:19:31: Buffer unpacking not optimized away.
warning: sfepy\discrete\common\extmods\_fmfield.pyx:32:31: Buffer unpacking not optimized away.
warning: sfepy\discrete\common\extmods\_fmfield.pyx:32:31: Buffer unpacking not optimized away.
warning: sfepy\discrete\common\extmods\_fmfield.pyx:45:31: Buffer unpacking not optimized away.
warning: sfepy\discrete\common\extmods\_fmfield.pyx:45:31: Buffer unpacking not optimized away.
warning: sfepy\discrete\common\extmods\_fmfield.pyx:58:28: Buffer unpacking not optimized away.
warning: sfepy\discrete\common\extmods\_fmfield.pyx:58:28: Buffer unpacking not optimized away.
warning: sfepy\discrete\common\extmods\_fmfield.pyx:66:28: Buffer unpacking not optimized away.
warning: sfepy\discrete\common\extmods\_fmfield.pyx:66:28: Buffer unpacking not optimized away.
warning: sfepy\discrete\common\extmods\_fmfield.pyx:73:29: Buffer unpacking not optimized away.
warning: sfepy\discrete\common\extmods\_fmfield.pyx:73:29: Buffer unpacking not optimized away.
warning: sfepy\discrete\common\extmods\_fmfield.pyx:81:29: Buffer unpacking not optimized away.
warning: sfepy\discrete\common\extmods\_fmfield.pyx:81:29: Buffer unpacking not optimized away.
error: Command "C:\Users\appveyor\AppData\Local\Programs\Common\Microsoft\Visual C++ for Python\9.0\VC\Bin\amd64\cl.exe /c /nologo /Ox /MD /W3 /GS- /DNDEBUG -D__SDIR__=\"sfepy\\mechanics\\extmods\" -DSFEPY_PLATFORM=1 -Dinline=__inline -Isfepy\mechanics\extmods -Isfepy\discrete\common\extmods -IC:\Python27_64\envs\sfepy-test\lib\site-packages\numpy\core\include -IC:\Python27_64\envs\sfepy-test\include -IC:\Python27_64\envs\sfepy-test\PC /Tcsfepy\mechanics\extmods\contres.c /Fobuild\temp.win-amd64-2.7\Release\sfepy\mechanics\extmods\contres.obj -g -O2" failed with exit status 2
```

**Comment by BubuLK** [comment]
BTW: IGAkit install/testing could be (easily) added to Appveryor config.

**Comment by rc** [comment]
@BubuLK Will you give it a shot? :)

**Comment by rc** [comment]
Thanks, @BubuLK , [42], [88] work, three to go...

**Comment by rc** [comment]
Solved for now by #495 .
